src/item.cpp

```diff
@@ -3921,60 +3921,39 @@ item *item::get_usable_item( const std::string &use_name )
 
 int item::pick_reload_ammo( const player &u, bool interactive )
 {
-    if( is_null() ) {
-        return INT_MIN;
-    }
-
-    if( !is_gun() && !is_tool() ) {
-        debugmsg( "RELOADING NON-GUN NON-TOOL" );
-        return INT_MIN;
-    }
-    int has_spare_mag = has_gunmod( "spare_mag" );
-    // Ammo types that we could use.
-    std::set<std::string> ammo_types;
-    // Specific item types that we could use (in case of partially loaded guns).
-    std::set<std::string> item_types;
-    if( is_gun() ) {
-        if( charges <= 0 && has_spare_mag != -1 && contents[has_spare_mag].charges > 0 ) {
-            // Special return to use magazine for reloading.
-            return INT_MIN + 1;
-        }
-
-        // If there's room to load more ammo into the gun or a spare mag, stash the ammo.
-        // If the gun is partially loaded make sure the ammo matches.
-        // If the gun is empty, either the spare mag is empty too and anything goes,
-        // or the spare mag is loaded and we're doing a tactical reload.
-        if( charges < clip_size() ||
-            ( has_spare_mag != -1 && contents[has_spare_mag].charges < spare_mag_size() ) ) {
-            if( charges > 0 && has_curammo() ) {
-                item_types.insert( get_curammo_id() );
+    std::set<std::string> ammo_types; // for unloaded items any ammo will do
+    std::set<std::string> item_types; // for partially loaded require matching type
+
+    // @todo deprecate handling of spare magazine as special case
+    auto spare = has_gunmod( "spare_mag" );
+    if( spare >= 0 ) {
+        if( ammo_remaining() <= 0 && contents[spare].charges > 0 ) {
+            return INT_MIN + 1; // special return to use spare magazine for reloading.
```

Comment:
`INT_MIN + 1` is inherited from the code handling the spare magazines which does raise the question as to how and when to deprecate this?